### PR TITLE
Wrap some objects in using statements

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: src/Foundation/NSDecimal.cs [hunk 1]
```diff
@@ -221,10 +229,7 @@ public bool Equals (NSDecimal other)
 
 		public override bool Equals (object obj)
 		{
-			if (!(obj is NSDecimal))
-				return false;
-			
-			return this == (NSDecimal) obj;
+			return obj is NSDecimal other && this == other;
```
Comment (chamons) [inline src/Foundation/NSDecimal.cs]: This is a cute one liner.

File: src/Security/Items.cs [hunk 2]
```diff
@@ -147,11 +147,12 @@ public static NSData QueryAsData (SecRecord query, bool wantPersistentReference,
 					if (max == 1)
 						return new NSData [] { new NSData (ptr, false) };
 
-					var array = new NSArray (ptr);
-					var records = new NSData [array.Count];
-					for (uint i = 0; i < records.Length; i++)
-						records [i] = new NSData (array.ValueAt (i), false);
-					return records;
+					using (var array = new NSArray (ptr)) {
```
Comment (chamons) [inline src/Security/Items.cs]: Any reason you used using blocks here but statements up above?

Comment (Therzok) [reply src/Security/Items.cs]: No reason at all. Switched to using statement.

**Comment by mandel-macaque** [comment]
build

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

**Pull request merged**